l10n_it_fatturapa_in/wizard/wizard_import_fatturapa.py

```diff
@@ -1183,6 +1183,21 @@ def _prepare_invoice_values(self, fatt, fatturapa_attachment, FatturaBody, partn
         self.set_e_invoice_lines(FatturaBody, invoice_data)
         return invoice_data
 
+    def _new_write_vals(self, invoice):
+        """
+        Update changing values only
+        """
+        to_return = invoice._convert_to_write(invoice._cache)
+        to_remove = set()
+        for k, val in to_return.items():
+            if (
+                k.endswith("_ids") and val[0][0] == 6 and val[0][2] == invoice[k].ids
+            ) or invoice[k] == val:
+                to_remove.add(k)
+        for k in to_remove:
+            del to_return[k]
```

Comment:
**suggestion**: `to_return` is a dictionary. We could probably do it all with a single loop using a dict comprehension, but I understand if you want to avoid it since it could become quite intricate.

Reply:
Exactly. At first I thought about using a dict comprehension but after implementing a small part of the checks it became quite hard to read